File: sorting/selection_sort.c
#include <stdio.h>
#include <stdlib.h>

// Sort array using selection sort algorithm
void selection_sort(int* array, size_t size){
    int tmp, min_index;
    for(int i = 0; i < (size - 1); i++){
        min_index = i;
        for(int j = (i + 1); j < size; j++){
            // Find index of min value
            if(array[j] < array[min_index])
                min_index = j;
        }
	// Move min value to the right place
        tmp = array[i];
        array[i] = array[min_index];
        array[min_index] = tmp;
    }
}

int main(){
    int array[] = {3, 2, -3, 9, 2, 0};
    for(int i = 0; i < 6; i++)
        printf("%d ", array[i]);

    selection_sort(array, 6);
    printf("\nSorted:\n");
    for(int i = 0; i < 6; i++)
        printf("%d ", array[i]);
    printf("\n");
}

File: sorting/selection_sort.py
from random import randint


def selection_sort(array: list) -> None:
    """Sorts array using selection sort algorithm"""

    for i in range(len(array)-1):
        min_index = i

        # Find index of min value
        for index, value in enumerate(array[i+1:]):
            if value < array[min_index]:
                # New min value
                min_index = index + (i + 1)

        # Swap values
        array[i], array[min_index] = array[min_index], array[i]


if __name__ == "__main__":
    # Generate random array
    array = [randint(0, 9) for _ in range(20)]

    print("Array: ")
    for i in array:
        print("{} ".format(i), end='')

    selection_sort(array)

    print("\nSorted array:")
    for i in array:
        print("{} ".format(i), end='')
    print("")
